torch/utils/tensorboard/_pytorch_graph.py

```diff
@@ -0,0 +1,267 @@
+import numpy as np
+import time
+from collections import OrderedDict
+
+from tensorboard.compat.proto.config_pb2 import RunMetadata
+from tensorboard.compat.proto.graph_pb2 import GraphDef
+from tensorboard.compat.proto.step_stats_pb2 import StepStats, DeviceStepStats, NodeExecStats, AllocatorMemoryUsed
+from tensorboard.compat.proto.versions_pb2 import VersionDef
+
+import torch
+from ._proto_graph import node_proto
+from torch.onnx.utils import OperatorExportTypes
+
+
+methods_OP = ['attributeNames', 'hasMultipleOutputs', 'hasUses', 'inputs',
+              'kind', 'outputs', 'outputsSize', 'scopeName']
+methods_IO = ['node', 'offset', 'uniqueName']  # 'unique' <int> , 'type' <Tensor<class 'torch._C.Type'>>
+
+
+class NodeBase(object):
+    def __init__(self, uniqueName=None, inputs=None, scope=None, tensor_size=None, op_type='UnSpecified', attributes=''):
+        self.uniqueName = uniqueName
+        self.inputs = inputs
+        self.tensor_size = tensor_size
+        self.kind = op_type
+        self.attributes = attributes
+        if scope is not None:
+            self.scope = scope
+
+    def __repr__(self):
+        repr = []
+        repr.append(str(type(self)))
+        for m in dir(self):
+            if '__' not in m:
+                repr.append(m + ': ' + str(getattr(self, m)) + str(type(getattr(self, m))))
+        return '\n'.join(repr) + '\n\n'
+
+
+class NodePy(NodeBase):
+    def __init__(self, NodeCpp, valid_methods):
+        super(NodePy, self).__init__(NodeCpp)
+        self.valid_methods = valid_methods[:]
+        self.inputs = []
+
+        for m in self.valid_methods:
```

Comment:
Call me a luddite, but I would have preferred it if we just boilerplated out each of the things we wanted to extract data out of, instead of metaprogramming it as it's done here.

Reply:
(With helper functions to reduce duplication.)

Reply:
@lanpa can you comment here? Thanks.